sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LogFilteringProcessor.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using OpenTelemetry.Logs;
+using OpenTelemetry;
+
+namespace Azure.Monitor.OpenTelemetry.AspNetCore
+{
+    internal class LogFilteringProcessor : BatchLogRecordExportProcessor
+    {
+        public LogFilteringProcessor(BaseExporter<LogRecord> exporter)
+            : base(exporter)
+        {
+        }
+
+        public override void OnEnd(LogRecord logRecord)
+        {
+            if (logRecord.SpanId == default || logRecord.TraceFlags == ActivityTraceFlags.Recorded)
```

Comment:
for my learning, why does the SpanId matter here?

Reply:
logs that are not emitted within an activity context will match this criteria.